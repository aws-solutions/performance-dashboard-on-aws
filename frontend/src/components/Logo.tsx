import React, { useState, useRef, useEffect } from "react";
import { useSettings, useLogo } from "../hooks";
import defaultLogo from "../logo.svg";
import { useTranslation } from "react-i18next";

function Logo() {
  const { t } = useTranslation();
  const { settings, loadingSettings } = useSettings();
  const { logo, loadingFile } = useLogo(settings.customLogoS3Key);
  const [toHide, setToHide] = useState<boolean>(true);

  // firstUpdate stops useEffect from executing after the first render.
  const firstUpdate = useRef(true);
  const secondUpdate = useRef(true);
  useEffect(() => {
    if (secondUpdate.current) {
      if (firstUpdate.current) {
        firstUpdate.current = false;
        return;
      }
      secondUpdate.current = false;
      return;
    }
    setToHide(false);
  }, [loadingFile]);

  return (
    <>
      {loadingFile || loadingSettings ? (
        <div />
      ) : (
        <>
          <img
            src={logo ? URL.createObjectURL(logo) : defaultLogo}
            alt={t("OrganizationLogo")}
            hidden={toHide}
          ></img>
        </>
      )}
    </>
  );
}

export default Logo;
