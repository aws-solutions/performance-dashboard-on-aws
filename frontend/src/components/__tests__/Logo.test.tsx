/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import Logo from "../Logo";

test("renders a logo", async () => {
    const wrapper = render(<Logo />);
    expect(wrapper.container).toMatchSnapshot();
});

test("renders the logo with the alt text", () => {
    const wrapper = render(<Logo />);
    const image = wrapper.getByTestId("logoImage");

    expect(image.getAttribute("alt")).toBe("Organization logo");
});
