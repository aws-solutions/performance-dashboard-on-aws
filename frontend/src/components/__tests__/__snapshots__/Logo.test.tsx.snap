// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a logo 1`] = `
<div>
  <img
    alt="Organization logo"
    hidden=""
    src="logo.svg"
  />
</div>
`;
